### PR TITLE
test(state): add tests

### PR DESCRIPTION
**Pull request by BioPhoton**

### CODE REVIEW

File: libs/state/spec/rx-state.service.spec.ts [hunk 1]
```diff
@@ -229,7 +263,7 @@ describe('RxStateService', () => {
   describe('connect', () => {
 
 
-    it('should add new slices', () => {
+    it('should work with observables directly', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/rx-state.service.spec.ts:266]:
```suggestion
    it('should take observable input', () => {
```

File: libs/state/spec/rx-state.service.spec.ts [hunk 2]
```diff
@@ -242,14 +276,59 @@ describe('RxStateService', () => {
       });
     });
 
-    it('should get previous state slices and accumulate', () => {
+    it('should work with prop name and observable', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/rx-state.service.spec.ts:279]:
```suggestion
    it('should connect an observable input to a property key', () => {
```

File: libs/state/spec/rx-state.service.spec.ts [hunk 3]
```diff
@@ -242,14 +276,59 @@ describe('RxStateService', () => {
...
+          a: 42,
+          b: 43,
+          c: 44
+        });
+
+        state.connect('num', from([{ num: 42 }, { num: 43 }, { num: 44 }]).pipe(map(s => s.num)));
+      });
+    });
+
+    it('should work with observable and project', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/rx-state.service.spec.ts:292]:
```suggestion
    it('should accept observable input and a projectFn', () => {
```

File: libs/state/spec/rx-state.service.spec.ts [hunk 4]
```diff
@@ -314,4 +393,33 @@ describe('RxStateService', () => {
     });
 
   });
+
+  describe('hold', () => {
+
+    it('should work with effect-observable', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/rx-state.service.spec.ts:399]:
```suggestion
    it('should stop the effect when input completes', () => {
```

File: libs/state/spec/rx-state.service.spec.ts [hunk 5]
```diff
@@ -314,4 +393,33 @@ describe('RxStateService', () => {
...
+        const sub = '(^!)';
+        const stop = new Subject();
+        state.hold(test$.pipe(takeUntil(stop)));
+        stop.next(1);
+        expectSubscriptions(test$.subscriptions).toBe(sub);
+
+      });
+    });
+
+    it('should work with observable and effect',  fakeAsync(() => {
```
Comment (hoebbelsB) [inline libs/state/spec/rx-state.service.spec.ts:413]:
```suggestion
    it('should accept observable input and an effect function',  fakeAsync(() => {
```

File: libs/state/src/lib/core/utils/typing.ts [hunk 6]
```diff
@@ -19,7 +19,7 @@ export function isPromiseGuard<T>(value: unknown): value is Promise<T> {
 export function isOperateFnArrayGuard<T, R = T>(
   op: any[]
 ): op is OperatorFunction<T, R>[] {
-  return op.length > 0 && op.every((i: any) => typeof i !== 'string');
+  return op.length > 0 && op.every((i: any) => typeof i === 'function');
```
Comment (hoebbelsB) [inline libs/state/src/lib/core/utils/typing.ts:22]: is typeof 'function' correct? isn't this a bit too generic?

Comment (BioPhoton) [reply libs/state/src/lib/core/utils/typing.ts:22]: An operator is a function.

We would have to put an observable in and see if one comes out to be really sure it is an operator.
😅😅

Comment (hoebbelsB) [reply libs/state/src/lib/core/utils/typing.ts:22]: bad idea :)

**Pull request merged**